netket/operator/_pauli_strings/jax.py

```diff
@@ -0,0 +1,355 @@
+# Copyright 2023 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import partial, wraps
+from typing import List, Union
+
+import numpy as np
+
+import jax
+from jax import numpy as jnp
+from jax.tree_util import register_pytree_node_class
+
+from netket.hilbert import AbstractHilbert, HomogeneousHilbert
+from netket.utils.types import DType
+
+from .._discrete_operator_jax import DiscreteJaxOperator
+
+from .base import PauliStringsBase
+
+from netket.operator._ising.jax import _ising_conn_states_jax
+
+# pauli-strings operator written in nJax
+# the general idea is the following:
+
+# observe that that Y = -i Z X
+# therefore for every pauli in a given operator we
+# (1.) apply X  (flip site)                  -- if it is X or Y
+# (2.) absorb -i to the weights              -- if it is Y
+# (3.) apply Z (pick up -1. if site is up/1) -- if it is Y or Z
+
+
+# TODO implement cutoff
```

Comment:
the base class supports the cutoff though? should we add an error message if somebody tries to use a cutoff with the jax version?

Reply:
ah ok sorry the check is already done